pallets/messages/src/tests.rs

```diff
@@ -238,226 +243,34 @@ fn on_initialize_should_add_messages_into_storage_and_clean_temp() {
-		assert_eq!(pagination_response.next_index, Some(from_index + page_size));
-
-		let payload = Vec::from("{'fromId': 123, 'content': '232323114432'}".as_bytes());
-		assert_eq!(
-			pagination_response.content[0],
-			MessageResponse {
-				msa_id: Some(10),
-				payload: Some(payload.clone()),
-				index: from_index as u16,
-				provider_msa_id: 1,
-				block_number: 0,
-				payload_length: None,
-				cid: None
-			}
-		);
-	});
-}
-
-#[test]
-fn get_messages_by_schema_with_valid_request_should_return_next_block_and_index() {
-	new_test_ext().execute_with(|| {
-		// arrange
-		let schema_id: SchemaId = 1;
-		let page_size = 7;
-		let from_index = 8;
-		let messages_per_block = vec![10, 0, 5, 2, 0, 3, 9];
-		populate_messages(schema_id, messages_per_block.clone(), PayloadLocation::OnChain);
-		let request = BlockPaginationRequest {
-			page_size,
-			from_block: 0,
-			to_block: messages_per_block.len() as u64,
-			from_index,
-		};
-
-		// act
-		let res = MessagesPallet::get_messages_by_schema(schema_id, request);
-
-		// assert
-		assert_ok!(&res);
-
-		let pagination_response = res.ok().unwrap();
-		assert_eq!(pagination_response.content.len() as u32, page_size);
-		assert_eq!(pagination_response.has_next, true);
-		assert_eq!(pagination_response.next_block, Some(3));
-		assert_eq!(pagination_response.next_index, Some(0));
-	});
-}
-
-#[test]
-fn get_messages_by_schema_with_less_messages_than_page_size_should_not_has_next() {
-	new_test_ext().execute_with(|| {
-		// arrange
-		let schema_id: SchemaId = 1;
-		let page_size = 30;
-		let from_index = 0;
-		let messages_per_block = vec![10, 0, 5, 2, 0, 3, 9];
-		populate_messages(schema_id, messages_per_block.clone(), PayloadLocation::OnChain);
-		let request = BlockPaginationRequest {
-			page_size,
-			from_block: 0,
-			to_block: messages_per_block.len() as u64,
-			from_index,
-		};
-
-		// act
-		let res = MessagesPallet::get_messages_by_schema(schema_id, request);
-
-		// assert
-		assert_ok!(&res);
-
-		let pagination_response = res.ok().unwrap();
-		assert_eq!(pagination_response.content.len() as u32, 29);
-		assert_eq!(pagination_response.has_next, false);
-		assert_eq!(pagination_response.next_block, None);
-		assert_eq!(pagination_response.next_index, None);
-	});
-}
-
-#[test]
-fn get_messages_by_schema_with_invalid_request_should_panic() {
-	new_test_ext().execute_with(|| {
-		// arrange
-		let schema_id: SchemaId = 1;
-		let page_size = 30;
-		let from_index = 0;
-		let messages_per_block = vec![10, 0, 5, 2, 0, 3, 9];
-		populate_messages(schema_id, messages_per_block, PayloadLocation::OnChain);
-		let request =
-			BlockPaginationRequest { page_size, from_block: 22, to_block: 15, from_index };
-
-		// act
-		let res = MessagesPallet::get_messages_by_schema(schema_id, request);
-
-		// assert
-		assert_err!(res, Error::<Test>::InvalidPaginationRequest);
-	});
-}
-
-#[test]
-fn get_messages_by_schema_with_overflowing_input_should_panic() {
```

Comment:
This was really hard to reproduce with the new structure of testing. This will likely need to move into an e2e test eventually.